scripts/install_submitter.py

```
@@ -0,0 +1,619 @@
+  - If dependency_bundle/ directory is empty or missing:
+    - Internet connection required for pip install commands
+    - PyPI access needed to download packages (deadline, xxhash, psutil, etc.)
+  - If dependency_bundle/*.zip files exist: Offline installation possible
+
+- RUNTIME ASSUMPTIONS:
+  - Single-user installation (not system-wide)
+  - VRED should be closed during installation for best results
+"""
+
+import argparse
+import base64
+import binascii
+import ctypes
+import logging
+import os
+import platform
+import shutil
+import sys
+import xml.etree.ElementTree as ET
+import zipfile
+from pathlib import Path
+from typing import List, Optional, TYPE_CHECKING
+
+# Import dependency bundle builder
+# Add the scripts directory to Python path for importing deps_bundle
+project_scripts_dir = Path(__file__).parent
+if str(project_scripts_dir) not in sys.path:
+    sys.path.insert(0, str(project_scripts_dir))
+
+try:
+    from deps_bundle import build_deps_bundle
+except ImportError as e:
+    print(f"Warning: Could not import deps_bundle: {e}")
+    build_deps_bundle = None
+
+if TYPE_CHECKING or platform.system().lower() == "windows":
+    try:
+        import winreg
+    except ImportError:
+        winreg = None  # type: ignore
+else:
+    winreg = None  # type: ignore
+
+logger = logging.getLogger(__name__)
+
+
+class VREDSubmitterInstaller:
+    """Handles complete installation and configuration for Deadline Cloud submitter in VRED Pro."""
+
+    def __init__(self):
+        self.is_windows = platform.system().lower() == "windows"
+        self.package_root = Path(__file__).parent.parent
+        print(f"Package root: {self.package_root}")
+
+    def get_default_install_directory(self) -> Path:
+        """Get the default installation directory based on VRED conventions."""
+        if not self.is_windows:
+            logger.warning("VRED Pro is only supported on Windows")
+            return Path.home() / ".deadline-cloud" / "vred-submitter"
```

If we only support windows, shouldn't we just throw an error in initialization and printing to the user that it only supports Windows at this time?

---

You're correct, I've moved the Windows check to the `__init__` method of `VREDSubmitterInstaller` class, so it fails fast when someone tries to run this script on a non-Windows.